File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/state/operation/RoomState.java
package com.trtc.uikit.livekit.state.operation;

import com.tencent.cloud.tuikit.engine.room.TUIRoomDefine;
import com.trtc.tuikit.common.livedata.LiveData;
import com.trtc.uikit.livekit.common.utils.Constants;
import com.trtc.uikit.livekit.state.LiveDefine;

import java.util.HashSet;
import java.util.Set;

public class RoomState {
    public String                           roomId        = "";
    public long                             createTime    = 0;
    public UserState.UserInfo               ownerInfo     = new UserState.UserInfo();
    public LiveData<String>                 roomName      = new LiveData<>("");
    public LiveData<String>                 coverURL      =
            new LiveData<>(Constants.DEFAULT_COVER_URL);
    public LiveData<TUIRoomDefine.SeatMode> seatMode      =
            new LiveData<>(TUIRoomDefine.SeatMode.FREE_TO_TAKE);
    public LiveData<Integer>                userCount     = new LiveData<>(0);
    public LiveData<Integer>                maxSeatCount  = new LiveData<>(0);
    public LiveExtraInfo                    liveExtraInfo = new LiveExtraInfo();

    public void reset() {
        createTime = 0;
        roomName.set("");
        coverURL.set(Constants.DEFAULT_COVER_URL);
        seatMode.set(TUIRoomDefine.SeatMode.FREE_TO_TAKE);
        userCount.set(0);
        maxSeatCount.set(0);
        liveExtraInfo = new LiveExtraInfo();
    }

    public void updateState(TUIRoomDefine.RoomInfo roomInfo) {
        this.roomId = roomInfo.roomId;
        this.createTime = roomInfo.createTime;
        this.roomName.set(roomInfo.name, false);
        this.seatMode.set(roomInfo.seatMode, false);
        this.ownerInfo.userId = roomInfo.ownerId;
        this.maxSeatCount.set(roomInfo.maxSeatCount, false);
    }

    public static class LiveExtraInfo {
        public LiveData<String>                             category         = new LiveData<>("");
        public LiveData<LiveDefine.LiveStreamPrivacyStatus> liveMode         =
                new LiveData<>(LiveDefine.LiveStreamPrivacyStatus.PUBLIC);
        public int                                          maxAudienceCount = 0;
        public int                                          messageCount     = 0;
        public int                                          giftIncome       = 0;
        public Set<String>                                  giftPeopleSet    = new HashSet<>();
        public int                                          likeCount        = 0;
    }
}

File: Flutter/lib/state/operation/room_state.dart
import 'package:flutter/material.dart';
import 'package:rtc_room_engine/api/room/tui_room_define.dart';
import 'package:tencent_live_uikit/common/constants/constants.dart';
import 'package:tencent_live_uikit/state/live_define.dart';
import 'package:tencent_live_uikit/state/operation/user_state.dart';

class RoomState with ChangeNotifier {
  String roomId = "";
  int createTime = 0;
  final UserInfo ownerInfo = UserInfo();
  final ValueNotifier<String?> roomName = ValueNotifier("");
  final ValueNotifier<String> coverURL = ValueNotifier(Constants.defaultCoverUrl);
  final ValueNotifier<TUISeatMode> seatMode = ValueNotifier(TUISeatMode.freeToTake);
  final ValueNotifier<int> userCount = ValueNotifier(0);
  final ValueNotifier<int> maxSeatCount = ValueNotifier(0);
  LiveExtraInfo liveExtraInfo = LiveExtraInfo();

  void reset() {
    createTime = 0;
    roomName.value = '';
    coverURL.value = Constants.defaultCoverUrl;
    seatMode.value = TUISeatMode.freeToTake;
    userCount.value = 0;
    maxSeatCount.value = 0;
    liveExtraInfo = LiveExtraInfo();
    notifyListeners();
  }

  void updateState(TUIRoomInfo? roomInfo) {
    if (roomInfo == null) {
      return;
    }
    roomId = roomInfo.roomId;
    createTime = roomInfo.createTime;
    roomName.value = roomInfo.name;
    seatMode.value = roomInfo.seatMode;
    ownerInfo.userId = roomInfo.ownerId;
    maxSeatCount.value = roomInfo.maxSeatCount;
    notifyListeners();
  }
}

class LiveExtraInfo {
  final ValueNotifier<int> category = ValueNotifier(0);
  final ValueNotifier<LiveStreamPrivacyStatus> liveMode = ValueNotifier(LiveStreamPrivacyStatus.publicity);
  final Set<String> giftPeopleSet = Set<String>();
  int maxAudienceCount = 0;
  int messageCount = 0;
  int giftIncome = 0;
  int likeCount = 0;
}
